code/datums/components/team_monitor.dm

```diff
@@ -0,0 +1,542 @@
+
+//===========
+// Handles the parent being moved and updates the direction of the arrows.
+//===========
+
+/datum/component/team_monitor/process()
+	update_all_directions()
+
+//When the parent is removed, we need to update our arrows
+//Also if we are visible update the arrows of anything tracking us
+/datum/component/team_monitor/proc/parent_moved()
+	//Update our alt appearences
+	update_all_directions()
+
+//Updates the direction of the arrows for all atoms we are tracking
+/datum/component/team_monitor/proc/update_all_directions()
+	if(!updating)
+		return
+	for(var/datum/component/tracking_beacon/beacon as() in tracking)
+		update_atom_dir(beacon)
+
+//Update the arrow towards another atom
+/datum/component/team_monitor/proc/update_atom_dir(datum/component/tracking_beacon/beacon)
+	if(!updating || !updating.hud_used || !beacon)
+		return
+	var/atom/movable/screen/arrow/screen = tracking[beacon]
+	var/turf/target_turf = get_turf(beacon.parent)
+	var/turf/parent_turf = get_turf(parent)
+	if(target_turf.z != parent_turf.z || target_turf == parent_turf)
+		if(screen)
+			//Remove the screen
+			if(updating.hud_used)
+				updating.hud_used.team_finder_arrows -= screen
+			qdel(screen)
+			tracking[beacon] = null
+			//Update their hud
+			if(updating.hud_used)
+				updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
+		return
+	if(!screen)
+		//Create the screen
+		screen = new
+		screen.alpha = 240
+		screen.color = beacon.colour
+		screen.hud = updating.hud_used
+		updating.hud_used.team_finder_arrows += screen
+		tracking[beacon] = screen
+		//Update their hud
+		updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
+	var/matrix/rotationMatrix = matrix()
+	rotationMatrix.Scale(1.5)
+	rotationMatrix.Translate(0, -distance)
+	rotationMatrix.Turn(Get_Angle(target_turf, parent_turf))
+	animate(screen, transform = rotationMatrix, time = 2)
+
+//===========
+// Handles being equipped / dequipped
+//===========
+
+//The parent equipped an item with a team_monitor, check if its in the right slot and apply the hud
+//Also needs to enable other trackers pointers towards us
+/datum/component/team_monitor/proc/parent_equipped(datum/source, mob/equipper, slot)
+	var/obj/item/clothing/item = parent
+	if(!istype(item))
+		return
+	if(item.slot_flags & slotdefine2slotbit(slot)) //Was equipped to a valid slot for this item?
+		show_hud(equipper)
+	else
+		hide_hud(equipper)
+
+//Disable our hud
+//Disable the pointers to us
+/datum/component/team_monitor/proc/parent_dequpped(datum/source, mob/user)
+	hide_hud(user)
+
+//===========
+// Handles hiding / showing the hud when equipped
+//===========
+
+/datum/component/team_monitor/proc/show_hud(mob/target)
+	updating = target
+	//Start processing to update in weird situations
+	START_PROCESSING(SSprocessing, src)
+	//Register parent signal
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/parent_moved)
+	//Our hud is disabled
+	if(!hud_visible)
+		return
```

Comment:
```suggestion
	updating = target
	//Our hud is disabled
	if(!hud_visible)
		return
	//Start processing to update in weird situations
	START_PROCESSING(SSprocessing, src)
	//Register parent signal
	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/parent_moved)
```

`hud_visible` disables hud, yet `update_all_directions` calls that'll be caused by `START_PROCESSING` and `RegisterSignal` will update & create arrows *without* checking `hud_visible`. Please consider the above suggestion, or add checks at `update_all_directions`